my_pyscf/mcpdft/laspdft.py

```diff
@@ -35,6 +37,65 @@ def make_casdm2s(filename, i):
     return rdm2s
 
 
+def get_energy_decomposition(mc, mo_coeff=None, ci=None, ot=None, otxc=None,
+                             grids_level=None, grids_attr=None,
+                             split_x_c=None, verbose=None):
+    """
+    I have to include this function to do the EDA for selected roots only.
+    """
+    if verbose is None: verbose = mc.verbose
+    log = lib.logger.new_logger(mc, verbose)
+    if mo_coeff is None: mo_coeff = mc.mo_coeff
+    if ci is None: ci = mc.ci
+    if grids_attr is None: grids_attr = {}
+    if grids_level is not None: grids_attr['level'] = grids_level
+    if len(grids_attr) or (otxc is not None):
+        old_ot = ot if (ot is not None) else mc.otfnal
+        old_grids = old_ot.grids
+        if otxc is None: otxc = old_ot.otxc
+        new_ot = get_transfnal(mc.mol, otxc)
+        new_ot.grids.__dict__.update(old_grids.__dict__)
+        new_ot.grids.__dict__.update(**grids_attr)
+        ot = new_ot
+    if ot is None: ot = mc.otfnal
+    if split_x_c is None:
+        split_x_c = True
+        log.warn('Currently, split_x_c in get_energy_decomposition defaults to '
+                 'True.\nThis default will change to False in the near future.')
+    hyb_x, hyb_c = ot._numint.hybrid_coeff(ot.otxc)
+    if hyb_x > 1e-10 or hyb_c > 1e-10:
+        raise NotImplementedError("Decomp for hybrid PDFT fnals")
+    if not isinstance(ot, transfnal):
+        raise NotImplementedError("Decomp for non-translated PDFT fnals")
+
+    if split_x_c:
+        ot = list(ot.split_x_c())
+    else:
+        ot = [ot, ]
+    e_nuc = mc._scf.energy_nuc()
+
+    nroots = getattr(mc.fcisolver, 'nroots', 1)
+    if not isinstance(nroots, list):
+        nroots = list(range(nroots))
+
+    if isinstance(nroots, list) and len(nroots) > 1:
```

Comment:
`lib.issequence (nroots)` is safer than `isinstance (nroots, list)` because the former will catch if it's a tuple, an ndarray, etc.